src/text/raw_text_reader.rs

```diff
@@ -1,476 +1,59 @@
-            {
-                // We found the unannotated zero that we appended to the end of the buffer.
-                // The "\n" in this pattern is the unparsed text left in the buffer,
-                // which indicates that our 0 was parsed.
-                RootParseResult::Eof
-            }
-            Ok((_remaining_text, value)) => {
-                // We found something else. The zero is still in the buffer; we can leave it there.
-                // The reader's `is_eof` flag has been set, so the text buffer will never be used
-                // again. Return the value we found.
-                RootParseResult::Ok(value)
-            }
-            Err(Incomplete(_needed)) => {
-                RootParseResult::Failure(format!(
-                    "Unexpected end of input on line {}: '{}'",
-                    self.buffer.lines_loaded(),
-                    &self.buffer.remaining_text()[..original_length] // Don't show the extra `\n0\n`
-                ))
-            }
-            Err(Error(ion_parse_error)) => {
-                RootParseResult::Failure(format!(
-                    "Parsing error occurred near line {}: '{}': '{:?}'",
-                    self.buffer.lines_loaded(),
-                    &self.buffer.remaining_text()[..original_length], // Don't show the extra `\n0\n`
-                    ion_parse_error
-                ))
+            let n = self.reader.read_from(&mut self.source, length)?;
+            bytes_read += n;
+            if n == 0 || bytes_read + n >= length {
+                break;
             }
-            Err(Failure(ion_parse_error)) => {
-                RootParseResult::Failure(format!(
-                    "A fatal error occurred while reading near line {}: '{}': '{:?}'",
-                    self.buffer.lines_loaded(),
-                    &self.buffer.remaining_text()[..original_length], // Don't show the extra `\n0\n`
-                    ion_parse_error
-                ))
-            }
-        };
-
-        // If we didn't consume the sentinel value, remove the sentinel value from the buffer.
-        // Doing so makes this method idempotent.
-        if self.buffer.remaining_text().ends_with(SENTINEL_ION_TEXT) {
-            let length = self.buffer.remaining_text().len();
-            self.buffer
-                .inner()
-                .truncate(length - SENTINEL_ION_TEXT.len());
         }
-
-        value
-    }
-
-    /// Constructs an [IonError::IllegalOperation] which explains that the reader was asked to
-    /// perform an action that is only allowed when it is positioned over the item type described
-    /// by the parameter `expected`.
-    fn expected<E: Display>(&self, expected: E) -> IonError {
-        illegal_operation_raw(format!(
-            "type mismatch: expected a(n) {} but positioned over a(n) {}",
-            expected,
-            self.current()
-        ))
+        self.bytes_fed += bytes_read;
```

Comment:
I can't remember, is this information not already tracked in the raw reader or text buffer?

Reply:
Yea, this was missed in the last update. I had removed the original use of `bytes_fed`, and the function that exposed it, but missed the actual field definition and update to it. Deleted.